[Reader] Only show/announce empty state when empty after updating posts

thomashorta opened the pull request:
Fixes #19452

The Reader fragment was showing the empty view (and asking for it to be announced on Talkback) for a fraction of a second before updating the "Fetching posts" message (on that same view). This caused talkback to say the message for an empty view, since the message update did not request an announcement to the user.

This was fixed by introducing a new flag that checks if the posts were actually updated from the backend to only show the empty view if we in fact have no posts.

Note: @hassaanelgarem I only included you as reviewer because you reported the issue but feel free to ignore it.

To test:
1. Enable talk back
2. Navigate to the reader
3. Open any post
4. Tap on any tag not previously opened
5. **Verify** talkback reads the correct state (fetching)

## Regression Notes
1. Potential unintended areas of impact
Showing/reading wrong feed state in other scenarios.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

Commit:
Only show/announce empty state when empty after updating posts

RenanLukas commented:
Thanks for the PR, @thomashorta

As discussed on DM, for me TalkBack is reading "Fetching posts" as expected but before that (right after opening the screen) it's reading "No posts with this topic".

The testing flow I went through was this:

1 - Discover
2 - Open post details
3 - Tap on a tag

Commit:
Fix Talkback saying "No posts..." while fetching (review comment)

Commit:
Fix "Fetching posts" text showing on top of posts

This was happening if there were already local DB posts being shown while the
fetching was happening, but that already shows a loading indicator so no need
for showing this "Fetching posts" text overlapping the posts.

RenanLukas reviewed (approved):
Thanks for the changes. It's working as expected :shipit:
